File: app/src/main/java/com/punchlines/punchline/paas/PaasRepository.java
package com.punchlines.punchline.paas;

import java.util.List;

import java8.util.concurrent.CompletableFuture;
import retrofit2.http.GET;
import retrofit2.http.Path;

import static com.punchlines.configuration.Configuration.ARTISTS;
import static com.punchlines.configuration.Configuration.ARTIST_ARGUMENT;
import static com.punchlines.configuration.Configuration.ARTIST_PUNCHLINES;
import static com.punchlines.configuration.Configuration.DAILY;
import static com.punchlines.configuration.Configuration.RANDOM;

public interface PaasRepository {

    @GET(DAILY)
    CompletableFuture<Punchline> dailyPunchline();

    @GET(RANDOM)
    CompletableFuture<Punchline> randomPunchline();

    @GET(ARTISTS)
    CompletableFuture<List<String>> artists();

    @GET(ARTIST_PUNCHLINES)
    CompletableFuture<List<Punchline>> artistPunchlines(@Path(ARTIST_ARGUMENT) String artist);

}

File: app/src/main/java/com/punchlines/punchline/paas/PaasRepository.kt
package com.punchlines.punchline.paas

import java8.util.concurrent.CompletableFuture
import retrofit2.http.GET
import retrofit2.http.Path

import com.punchlines.application.Configuration.ARTISTS
import com.punchlines.application.Configuration.ARTIST_ARGUMENT
import com.punchlines.application.Configuration.ARTIST_PUNCHLINES
import com.punchlines.application.Configuration.DAILY
import com.punchlines.application.Configuration.RANDOM

interface PaasRepository {

    @GET(DAILY)
    fun dailyPunchline(): CompletableFuture<Punchline>

    @GET(RANDOM)
    fun randomPunchline(): CompletableFuture<Punchline>

    @GET(ARTISTS)
    fun artists(): CompletableFuture<List<String>>

    @GET(ARTIST_PUNCHLINES)
    fun artistPunchlines(@Path(ARTIST_ARGUMENT) artist: String): CompletableFuture<List<Punchline>>

}